server/src/main/java/org/opensearch/action/admin/indices/tiering/TransportHotToWarmTieringAction.java

```diff
@@ -0,0 +1,126 @@
+import org.opensearch.cluster.routing.allocation.DiskThresholdSettings;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.index.Index;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.opensearch.indices.tiering.TieringRequestValidator.validateHotToWarm;
+
+/**
+ * Transport Tiering action to move indices from hot to warm
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class TransportHotToWarmTieringAction extends TransportClusterManagerNodeAction<TieringIndexRequest, HotToWarmTieringResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportHotToWarmTieringAction.class);
+    private final ClusterInfoService clusterInfoService;
+    private final DiskThresholdSettings diskThresholdSettings;
+
+    @Inject
+    public TransportHotToWarmTieringAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        ClusterInfoService clusterInfoService,
+        Settings settings
+    ) {
+        super(
+            HotToWarmTieringAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            TieringIndexRequest::new,
+            indexNameExpressionResolver
+        );
+        this.clusterInfoService = clusterInfoService;
+        this.diskThresholdSettings = new DiskThresholdSettings(settings, clusterService.getClusterSettings());
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected HotToWarmTieringResponse read(StreamInput in) throws IOException {
+        return new HotToWarmTieringResponse(in);
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(TieringIndexRequest request, ClusterState state) {
+        ClusterBlockException blockException = state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
+        if (blockException == null) {
+            for (String index : request.indices()) {
+                try {
+                    blockException = state.blocks()
+                        .indicesBlockedException(
+                            ClusterBlockLevel.METADATA_WRITE,
+                            indexNameExpressionResolver.concreteIndexNames(clusterService.state(), request.indicesOptions(), true, index)
```

Comment:
if a specific index have write blocked, should we just add that index in list of failed tiering indices, and still proceed with other indices?

Reply:
that depends on the IndicesOptions specified in the request and concreteIndexNames function takes care of handling that